src/frequenz/sdk/actor/_run_utils.py

```diff
@@ -38,18 +38,22 @@ async def run(*actors: Actor) -> None:
         done_tasks, pending_tasks = await asyncio.wait(
             pending_tasks, return_when=asyncio.FIRST_COMPLETED
         )
-
         # This should always be only one task, but we handle many for extra safety
         for task in done_tasks:
-            # Cancellation needs to be checked first, otherwise the other methods
-            # could raise a CancelledError
-            if task.cancelled():
+            # BackgroundService returns a BaseExceptionGroup containing multiple
+            # exceptions. The 'await' statement raises these exceptions, and 'except*'
+            # is used to handle them as a group. If the task raises multiple different
+            # exceptions, 'except*' will be invoked multiple times, once for each
+            # exception group.
+            try:
+                await task
```

Comment:
I'm thinking that in this case it might be better to just call [`task.result()`](https://docs.python.org/3/library/asyncio-task.html#asyncio.Task.result) so we avoid the `await` (entering the async loop) when we know the task is finished anyway and we don't need to `await` it.